source/fundamentals/connection/connect.txt

```diff
@@ -0,0 +1,123 @@
+     - Required. The host and optional port number where MongoDB is running. If you don't include the port number, the driver will use the default port, ``27017``.
+
+   * - ``/defaultauthdb``
+     
+     - Optional. The authentication database to use if the
+       connection string includes ``username:password@``
+       authentication credentials but not the ``authSource`` option. If you don't include 
+       this piece, the client will authenticate the user against the ``admin`` database.
+
+   * - ``?<options>``
+
+     - Optional. A query string that specifies connection-specific
+       options as ``<name>=<value>`` pairs. See
+       :ref:`csharp-connection-options` for a full description of
+       these options.
+
+To use a connection URI, pass it as a string to the ``MongoClient`` constructor. In the 
+following example, the driver uses a sample connection URI to connect to a MongoDB 
+instance on port ``27017`` of ``localhost``: 
+
+.. literalinclude:: /includes/fundamentals/code-examples/connection/LocalConnection.cs
+   :language: csharp
+
+.. tip::
+
+    See :manual:`the MongoDB Manual</reference/connection-string>`: for more information about creating a connection string.
+
+MongoClientSettings
+-------------------
+
+You can use a ``MongoClientSettings`` object to configure the connection in code
+rather than in a connection URI. To use a ``MongoClientSettings`` object, create an 
+instance of the class and pass it as an argument to the ``MongoClient`` constructor. 
+
+In the following example, the driver uses a ``MongoClientSettings`` object to connect to a 
+MongoDB instance on port ``27017`` of ``localhost``:
+
+.. literalinclude:: /includes/fundamentals/code-examples/connection/MongoClientSettings.cs
+   :language: csharp
+   :dedent:
+
+Other Connection Targets
+------------------------
+
+Connect to a Standalone Deployment
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+In the following example, the driver uses a sample connection URI to connect to a MongoDB instance on port ``27017`` of ``sample.host`` with the credentials ``user1`` and ``password1``:
+
+.. literalinclude:: /includes/fundamentals/code-examples/connection/StandaloneConnection.cs
+   :language: csharp
+
+Connect to Atlas
+~~~~~~~~~~~~~~~~
+
+In the following example, the driver uses a sample connection URI to connect to a MongoDB instance on Atlas with the credentials ``user1`` and ``password1``:
+
+.. literalinclude:: /includes/fundamentals/code-examples/connection/AtlasConnection.cs
+   :language: csharp
+
+.. tip::
+
+   If your deployment is hosted on Atlas, follow the
+   :atlas:`Atlas driver connection guide </driver-connection?tck=docs_driver_nodejs>`
+   to retrieve your connection string.
+
+Connect to a Replica Set
+~~~~~~~~~~~~~~~~~~~~~~~~
+
+In the following example, the driver uses a sample connection URI to connect to the MongoDB replica set ``sampleRS``, which is running on port ``27017`` of ``sample.host1``, ``sample.host2``, and ``sample.host3``:
```

Comment:
this is a bit confusing. It can be read like if you want to use rs connection string, you should provide all servers in the connection string which is not true. I think the main using mode is providing a single any rs host and `replicaSet` name